src/legacy/ui/public/vislib/visualizations/point_series/column_chart.js

```diff
@@ -236,13 +304,41 @@ export class ColumnChart extends PointSeries {
       return Math.abs(yScale(baseValue) - yScale(d.y));
     }
 
+    function formatValue(d) {
+      return chartData.yAxisFormatter(d.y);
+    }
+
     // update
     bars
       .attr('x', isHorizontal ? x : y)
       .attr('width', isHorizontal ? widthFunc : heightFunc)
       .attr('y', isHorizontal ? y : x)
       .attr('height', isHorizontal ? heightFunc : widthFunc);
 
+    const layer = d3.select(bars[0].parentNode);
+    const barLabels = layer.selectAll('text').data(chartData.values.filter(function (d) {
+      return !_.isNull(d.y);
+    }));
+
+    barLabels
+      .exit()
+      .remove();
+
+    if (isLabels) {
+      const labelColor = this.handler.data.getColorFunc()(chartData.label);
+
+      barLabels
+        .enter()
+        .append('text')
+        .text(formatValue)
+        .attr('font-size', '8pt')
+        .attr('x', isHorizontal ? labelX : labelY)
+        .attr('y', isHorizontal ? labelY : labelX)
+        .attr('dominant-baseline', isHorizontal ? 'auto' : 'central')
+        .attr('text-anchor', isHorizontal ? 'middle' : 'start')
+        .attr('fill', labelColor)
+        .attr('display', labelDisplay);
```

Comment:
You should add the `pointer-events: none;` because the label blocks the bar from getting the mouseover event to display the tooltip. This is more visible with smaller bars where the label text cover the most part of the bar.